deliveries/starlay_finance-milestone_1.md

```diff
@@ -0,0 +1,20 @@
+# Milestone Delivery :mailbox:
+
+**The invoice has been send out correctly for this milestone and the delivery is according to the official [milestone delivery guidelines](https://github.com/smart-contract-bounty/Support-Docs/blob/master/milestone-deliverables-guidelines.md).**  
+
+* **Application Document:** [Starlay Finance](https://github.com/Chidasan/Ecosystem-Grants/blob/master/applications/starlay_finance.md).
+
+**Deliverables**
+
+| Number | Deliverable | Link | Notes |
+| ------------- | ------------- | ------------- |------------- |
+| 0a. | License | [License](https://github.com/starlay-finance/starlay-protocol-wasm/blob/main/LICENSE.md) | 　|
+| 0b. | Documentation | [Documentation](https://docs.starlay.finance) |　 |
+| 1. | Deposit function | [Deposit function](https://github.com/starlay-finance/starlay-protocol-wasm/blob/main/logics/impls/pool/mod.rs#L308)  | |
+| 2. | Borrow function | [Borrow function](https://github.com/starlay-finance/starlay-protocol-wasm/blob/main/logics/impls/pool/mod.rs#L339) | |
+| 3. | Redeem function | [Redeem function](https://github.com/starlay-finance/starlay-protocol-wasm/blob/main/logics/impls/pool/mod.rs#L322) | |
+| 4. | Repay_borrow function| [Repay_borrow function](https://github.com/starlay-finance/starlay-protocol-wasm/blob/main/logics/impls/pool/mod.rs#L35)
```

Comment:
Can you add GitHub permalinks here instead? Or have a tag for each milestone submission? As you push more code to the repo these links get out of sync and don't line up anymore

Reply:
Thank you for your insightful review. We will carefully consider each of your points and respond accordingly. However, some responses may be challenging due to constraints related to time and manpower. We’d appreciate your understanding as we navigate through these issues, and would like to propose addressing certain parts in the next milestone.

>You’re using OpenBrush’s ZERO_ADDRESS in a lot of places. I’d recommend steering away from this and using something like Rust’s Option instead.

There are some security implications of using the zero address since it has a known private key. We understand the vulnerabilities. https://github.com/paritytech/ink/pull/1255 It’s just something we haven’t released, so I’d like to make it a required item for the next milestone.

>You should be running the TypeScript E2E tests as part of your CI. You should make use of the ink_e2e crate to do E2E testing in Rust. This would be good to ensure that functionality works independent of the TS tests.

Since we are currently using swanky-node, we need to change the infrastructure of E2E itself. The TS E2E itself is already in place, and the quality implementation itself is at least satisfactory, so incorporation into CI and selection of tools to be used is a type of improvement. Let this also be a necessary element of the next milestone.

In conclusion, while we intend to address some of the points mentioned, we will provide you with an update on our progress by the end of this week. Please note that some of the issues highlighted might not be addressed in the immediate term due to the constraints mentioned earlier. However, we aspire to tackle them in the forthcoming milestone as much as possible.